lib/utils.js

```diff
@@ -203,6 +203,17 @@ var utils = module.exports = {
 
     return obj;
   },
+
+  /**
+  * Determine if file data is a derivative of EventEmitter class.
+  * https://nodejs.org/api/events.html#events_events
+  */
+  checkForStream: function (obj) {
+    if (obj.file && obj.file.data) {
+      return typeof obj.file.data.on === 'function' && typeof obj.file.data.emit === 'function';
```

Comment:
Why not do an `instanceof` check? This is probably fine too, just curious.

Reply:
@rattrayalex-stripe good question! An `instanceof` check would have been more explicit, but I'm not aware of `require('events').EventEmitter` available in the utils file. In retrospect, I was trying to implement the feature with just minimal additions to the existing code. When it comes to collaboration, though, it makes sense to be more explicit - I should have probably just added it! Thanks for the feedback. I plan to refactor the `requestDataProcessor` function to async, and if I need this check, I'll implement via `instanceof `. Let me know if you have any other feedback - thanks!

Reply:
Cool, thanks! That reasoning makes sense. A duck-type check as you have here may make more sense; let's see how it looks after the refactor. Cheers